File: warmup/week02/day02_scrabble_score/js/scrabble.js
console.log('run');

// Version 1 (this way)
/*
const scrabbleScore = {
  letterValues: {
    1: ["A", "E", "I", "O", "U", "L", "N", "R", "S", "T" ],
    2: ["D", "G"],
    3: ["B", "C", "M", "P"],
    4: ["F", "H", "V", "W", "Y"],
    5: ["K"],
    8: ["J", "X"],
    10: ["Q", "Z"]
  },

  wordScore: function (word) {
    word = word.toUpperCase();
    let sum = 0;
    for (let i = 0; i < word.length; i++) {
      for (let score in this.letterValues) {
        if (this.letterValues[score].includes(word[i])) {
          sum += Number(score);
        }
      }
    }
    return sum;
  }
};

console.log(scrabbleScore.wordScore("cAbbAge"));
*/

/*
const letterValues = {
  1: ["A", "E", "I", "O", "U", "L", "N", "R", "S", "T" ],
  2: ["D", "G"],
  3: ["B", "C", "M", "P"],
  4: ["F", "H", "V", "W", "Y"],
  5: ["K"],
  8: ["J", "X"],
  10: ["Q", "Z"]
};

const letterScore = function (letter) {
  let points = 0;
  for (let score in letterValues) {
    if (letterValues[score].indexOf(letter) > -1) {
      points += Number(score);
    }
  }
  return points;
}

const scrabbleScore = function (word) {
  word = word.toUpperCase();
  let sum = 0;
  for (let i = 0; i < word.length; i++) {
    sum += letterScore(word[i]);
  }
  return sum;
}

console.log(scrabbleScore("cAbbAge"));

*/

//Version 3
/*
const scrabbleScore = {
  letterValues: {
      'a' : 1, 'e' : 1, 'i' : 1, 'o' : 1,
      'u' : 1, 'l' : 1, 'n' : 1, 'r' : 1,
      's' : 1, 't' : 1, 'd' : 2, 'g' : 2,
      'b' : 3, 'c' : 3, 'm' : 3, 'p' : 3,
      'f' : 4, 'h' : 4, 'v' : 4, 'w' : 4,
      'y' : 4, 'k' : 5, 'j' : 8, 'x' : 8,
      'q' : 10, 'z' : 10
    },

  wordScore: function (word) {
    word = word.toLowerCase();
    let sum = 0;
    for (let i = 0; i < word.length; i++) {
      sum += this.letterValues[word[i]];
    }
    return sum;
  }
};

console.log(scrabbleScore.wordScore("cabbage"));
*/

letterValues = {
  1: ["A", "E", "I", "O", "U", "L", "N", "R", "S", "T" ],
  2: ["D", "G"],
  3: ["B", "C", "M", "P"],
  4: ["F", "H", "V", "W", "Y"],
  5: ["K"],
  8: ["J", "X"],
  10: ["Q", "Z"]
  };

  const invert = function (input) {

    let invertedObject = {};

    for (let keys in input) {
      console.log('keys', keys);
      console.log('values', input[keys]);
      invertedObject[input[keys]] = keys
      console.log('invertedObject', invertedObject);
    }
    //return invertedObject;
  };

  invert(letterValues);

File: warmup/week04/day03_ruby_scrabble/scrabble.rb
# version 1

# def scrabble_score word
#
#   letter_values = {
#     1 => ["A", "E", "I", "O", "U", "L", "N", "R", "S", "T"],
#     2 => %w{D G},
#     3 => %w{B C M P},
#     4 => %w{F H V W Y},
#     5 => %w{K},
#     8 => %w{J X},
#     10 => %w{Q Z}
#   }
#
#   score = 0
#
#   word.upcase.each_char do |w|
#     letter_values.each do |key, value|
#       #puts "Key: #{key}, Value: #{value}"
#       if value.include? w
#         score += key
#       end
#     end
#   end
#
#   puts "The scrabble score for #{word} is #{score}"
#
# end
#
# scrabble_score "cabbage"


# Version 2

def scrabble_score word

  letter_values = {
    "A"=>1, "E"=>1, "I"=>1, "O"=>1, "U"=>1, "L"=>1, "N"=>1, "R"=>1, "S"=>1, "T"=>1, "D"=>2, "G"=>2, "B"=>3, "C"=>3, "M"=>3, "P"=>3, "F"=>4, "H"=>4, "V"=>4, "W"=>4, "Y"=>4, "K"=>5, "J"=>8, "X"=>8, "Q"=>10, "Z"=>10
    }

    score = 0

    word.upcase.each_char do |w|
      score += letter_values[w]
    end

    score

end

print "Type the word: "

input = gets.chomp

puts "The score for the word #{input} is #{scrabble_score input}"
